app/utils.ts

```diff
@@ -63,3 +64,29 @@ export function useUser(): User {
   }
   return maybeUser;
 }
+
+/**
+ * This async Fn takes a BullMQ job, and the queue name
+ * and returns children jobs that are of that queue name
+ * @param {string} jobName
+ * @param {Job} job
+ *
+ * @returns {Promise<Object>}
+ */
+
+export async function getChildrenValuesOfQueueName<CT>({
+  queueName,
+  job,
+}: {
+  queueName: string;
+  job: Job;
+}): Promise<{
+  [jobKey: string]: CT;
+}> {
+  const childrenValues = await job.getChildrenValues();
+  const filteredChildrenValues = Object.keys(childrenValues)
+    .filter((jobKey) => jobKey.includes(queueName))
+    .reduce((acc, jobKey) => ({ ...acc, [jobKey]: childrenValues[jobKey] }), {});
```

Comment:
I'm not sure if I understand correctly. Is this assuming that multiple values can be returned?

Reply:
In theoriy, you could have multiple of the same child processes, so we are selecting all of those that match, later we can get the [0] element if there has to be a single one